substrate/runtime/timestamp/src/lib.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright 2017 Parity Technologies (UK) Ltd.
+// This file is part of Substrate Demo.
+
+// Substrate Demo is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate Demo is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate Demo.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Timestamp manager: just handles the current timestamp.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[cfg_attr(test, macro_use)] extern crate substrate_runtime_std as rstd;
+#[macro_use] extern crate substrate_runtime_support as runtime_support;
+#[cfg(test)] extern crate substrate_runtime_io as runtime_io;
+extern crate substrate_codec as codec;
+
+#[cfg(feature = "std")] #[macro_use] extern crate serde_derive;
+#[cfg(feature = "std")] extern crate serde;
+
+use runtime_support::storage::StorageValue;
+use runtime_support::PublicPass;
+
+pub trait Trait {
+	type Timestamp: codec::Slicable;
+	type PublicAux;
+	type PrivAux;
```

Comment:
it seems to me that the only thing which is actually variable here is the timestamp type definition and that the others should follow, while the definition of these `PublicAux` and `PrivAux` as associated types implies that it's expected for them to vary while the timestamp type holds.